docs/abi.md

```diff
@@ -87,17 +87,19 @@ functions** as
 ### wait_on_channels
 
 `wait_on_channels(usize, u32) -> u32` blocks until data is available for reading
-from one of the specified channel handles. The channel handles are encoded in a
-buffer that holds N contiguous 9-byte chunks, each of which is made up of an
-8-byte channel handle value (little-endian u64) followed by a single channel
-status byte. Invalid handles will have an `INVALID_CHANNEL` status, but
+from one of the specified channel handles, unless any of the channels is
+invalid, orphaned, or violates the
+[information flow control](/docs/concepts.md#labels). The channel handles are
+encoded in a buffer that holds N contiguous 9-byte chunks, each of which is made
+up of an 8-byte channel handle value (little-endian u64) followed by a single
+channel status byte. Invalid and orphaned handles will have an `INVALID_CHANNEL`
```

Comment:
I would still make it very clear that the overall return value in this case is still `OK`, so callers need to be careful about checking the individual channel statuses (which I guess is fair enough because in general they would want to know which channel - if any - actually got ready).

Reply:
The sentence after this states the condition for `error`, which I think is clear enough. Nevertheless, I have reworded this sentence to look more like the original version.